app/src/main/java/com/dessalines/thumbkey/ui/screens/AbbreviationsScreen.kt

```diff
@@ -0,0 +1,226 @@
+                        onEdit = { abbreviationToEdit = it },
+                        onDelete = { abbreviationViewModel.delete(it.id) },
+                    )
+                }
+            }
+        }
+
+        if (showAddDialog) {
+            AddEditAbbreviationDialog(
+                abbreviation = null,
+                onDismiss = { showAddDialog = false },
+                onSave = { abbr, expansion ->
+                    abbreviationViewModel.insertOrUpdate(abbr, expansion)
+                    showAddDialog = false
+                },
+            )
+        }
+
+        abbreviationToEdit?.let { abbreviation ->
+            AddEditAbbreviationDialog(
+                abbreviation = abbreviation,
+                onDismiss = { abbreviationToEdit = null },
+                onSave = { abbr, expansion ->
+                    abbreviationViewModel.insertOrUpdate(abbr, expansion, abbreviation.id)
+                    abbreviationToEdit = null
+                },
+            )
+        }
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun AbbreviationItem(
+    abbreviation: Abbreviation,
+    onEdit: (Abbreviation) -> Unit,
+    onDelete: (Abbreviation) -> Unit,
+) {
+    Card(
+        modifier =
+            Modifier
+                .fillMaxWidth()
+                .padding(vertical = 4.dp),
+        onClick = { onEdit(abbreviation) },
+    ) {
+        Row(
+            modifier =
+                Modifier
+                    .padding(16.dp)
+                    .fillMaxWidth(),
+            horizontalArrangement = Arrangement.SpaceBetween,
+            verticalAlignment = Alignment.CenterVertically,
+        ) {
+            Column {
+                Text(
+                    text = abbreviation.abbreviation,
+                    style = MaterialTheme.typography.titleMedium,
+                )
+                Text(
+                    text = abbreviation.expansion,
+                    style = MaterialTheme.typography.bodyMedium,
+                )
+            }
+            IconButton(onClick = { onDelete(abbreviation) }) {
+                Icon(Icons.Default.Delete, contentDescription = "Delete")
+            }
+        }
+    }
+}
+
+@Composable
+fun AddEditAbbreviationDialog(
+    abbreviation: Abbreviation?,
+    onDismiss: () -> Unit,
+    onSave: (String, String) -> Unit,
+) {
+    var abbr by remember { mutableStateOf(abbreviation?.abbreviation ?: "") }
+    var expansion by remember { mutableStateOf(abbreviation?.expansion ?: "") }
+
+    AlertDialog(
+        onDismissRequest = onDismiss,
+        title = {
+            Text(
+                text =
+                    if (abbreviation == null) {
+                        stringResource(R.string.add_abbreviation)
+                    } else {
+                        stringResource(R.string.edit_abbreviation)
+                    },
+            )
+        },
+        text = {
+            Column {
```

Comment:
Instead of adding paddings below, it'd probably be better to use `verticalArrangement = Arrangement.spacedBy(...`